R/replace.R

```diff
@@ -80,6 +87,13 @@ replace_special_keywords <- function(expr_quotes_masked) {
     }
   }
 
+  # replace TRY_CONVERT
+  if (grepl(paste0("\\bTRY_CONVERT\\b"), expr_quotes_masked, ignore.case = TRUE)) {
+    # TRY_CONVERT converts to NULL when attempt fails, just like R's `as.*` functions
+    # So, we can handle it the same as CONVERT
```

Comment:
Do you think we should enclose the `as.*` function call in `suppressWarnings()` in the case of `TRY_CONVERT`?

Reply:
That's a good call. Should `CAST` and `CONVERT` throw an error on failed conversions? Currently, they just warn "NAs introduced by coercion".

Reply:
I think `CAST` and `CONVERT` should _not_ throw errors on failed conversions; in many SQL engines, they don't.

I think for `CAST` and `CONVERT`, the existing behavior (giving a warning) is best; and for `TRY_CAST` and `TRY_CONVERT`, silently returning would be best.